### PR TITLE
Added Autoparallel optimizer in config.py

### LINKED ISSUE #49507: No enable autoparallel optimizer doc in grappler
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/config/optimizer/set_experimental_options#args

## Description of issue (what needs changing):

The [document of grappler ](https://www.tensorflow.org/guide/graph_optimization#available_graph_optimizers) says it has an `autoparallel optimizer`, and it supposed to be enabled using `tf.config.optimizer.set_experimental_options`.  While in [`tf.config.optimizer.set_experimental_options`](https://www.tensorflow.org/api_docs/python/tf/config/optimizer/set_experimental_options#args), it does not have the corresponding option (auto_parallel).

### PR DESCRIPTION
**Pull request by SuryanarayanaY**
Autoparallel optimizer available as one of the Graph Optimizer as per [documentation](https://www.tensorflow.org/guide/graph_optimization#available_graph_optimizers) but for `tf.config.optimizer.set_experimental_options` it is not listed in the Arg options.But as the per this [comment](https://discuss.tensorflow.org/t/enable-grapplers-autoparallel-pass/1084/2) it is already available but not documented
Hence made a PR as Doc Bug.
Fixes #49507

### CODE REVIEW

File: tensorflow/python/framework/config.py [hunk 1]
```diff
@@ -237,6 +237,7 @@ def set_optimizer_experimental_options(options):
       - disable_meta_optimizer: Disable the entire meta optimizer.
       - min_graph_nodes: The minimum number of nodes in a graph to optimizer.
         For smaller graphs, optimization is skipped.
+      - Autoparallel optimizer - Automatically parallelizes graphs by splitting along the batch dimension
```
Comment (kenfranko) [inline tensorflow/python/framework/config.py:240]: This needs to match the Gappler name so it should be auto_parallel - Automatically ....  See https://github.com/tensorflow/tensorflow/blob/26949be08531e29eba6291053b3a679a1663036f/tensorflow/core/grappler/optimizers/meta_optimizer.cc#L247

File: tensorflow/python/framework/config.py [hunk 2]
```diff
@@ -237,7 +237,7 @@ def set_optimizer_experimental_options(options):
       - disable_meta_optimizer: Disable the entire meta optimizer.
       - min_graph_nodes: The minimum number of nodes in a graph to optimizer.
         For smaller graphs, optimization is skipped.
-      - Autoparallel optimizer - Automatically parallelizes graphs by splitting along the batch dimension
+      - auto_parallel : Automatically parallelizes graphs by splitting along the batch dimension
```
Comment (kenfranko) [inline tensorflow/python/framework/config.py:240]: Remove the space after auto_parallel for consistency.  Split across two lines, see min_graph_nodes as an example.  Max line length is 80 characters and this current change fails the linter.

**Review by kenfranko — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@SuryanarayanaY Can you please take a look on the above comment from @kenfranko. Thank you!

File: tensorflow/python/framework/config.py (hunk 2 above)
Comment (SuryanarayanaY) [reply tensorflow/python/framework/config.py]: Changes done as required.Thank you!

**Review by kenfranko — APPROVED** [review]

**Pull request merged**